src/gobupload/update/event_applicator.py

```diff
@@ -140,12 +144,18 @@ def apply(self, event):
         count = 1
         applied_events = []
 
+        entity_source_id = data.get('_entity_source_id')
+
         if isinstance(gob_event, GOB.BULKCONFIRM):
             self.storage.bulk_update_confirms(gob_event, event.eventid)
             count = len(gob_event._data['confirms'])
-        elif isinstance(gob_event, GOB.ADD) and self.last_events.get(data["_entity_source_id"]) is None:
+        elif isinstance(gob_event, GOB.ADD) and self.last_events.get(entity_source_id) is None and \
+                entity_source_id not in self.add_event_source_ids:
             # Initial add (an ADD event can also be applied on a deleted entity, this is handled by the else case)
             applied_events += self.add_add_event(gob_event)
+
+            # Store the entity_source_id to make sure a second ADD events get's handled as an ADD on deleted entity
+            self.add_event_source_ids.add(data.get('_entity_source_id'))
```

Comment:
.. and here

Maybe inline the first one in the elif expression, or use the variable here (second is probably better).

Reply:
Ah, yes placed it above to be able to use the variable. But then I should use it ofcourse...